File: backend/bluffword-backend/src/main/java/org/bluffwordbackend/controllers/CategoryController.java
package org.bluffwordbackend.controllers;

import lombok.RequiredArgsConstructor;
import org.bluffwordbackend.services.WordPairService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/categories")
@RequiredArgsConstructor
public class CategoryController {

    private final WordPairService wordPairService;

    @GetMapping
    public ResponseEntity<List<String>> getAllWordCategories() {

        return ResponseEntity.ok(wordPairService.getWordCategories());
    }
}

File: backend_kotlin/src/main/kotlin/org/project/backend_kotlin/Category/CategoryController.kt
package org.project.backend_kotlin.Category

import org.project.backend_kotlin.wordPair.CategoryRepository
import org.project.backend_kotlin.wordPair.WordPairService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/api/categories")
@RestController
class CategoryController(
    private val wordPairService: WordPairService
) {


    @GetMapping()
    fun getAllCategories(): ResponseEntity<List<String>> {
        return  ResponseEntity.ok(wordPairService.getAllCategories())
    }
}